build/stage1: support local systemd source for offline builds

dm0- opened the pull request:
Local systemd source trees are useful here for packagers to build usr_from_src without requiring network access.  This differentiates between local and remote systemd sources by checking if the value from `./configure` contains "://".  If the string is found, the build will preserve existing behavior.  If not, the value is interpreted as a direct specification of the source directory path.

lucab commented:
@fabiokung would you mind a review and perhaps a quick test that it doesn't break your setup?

fabiokung commented:
@lucab will do.

Meanwhile, I've been using `--with-stage1-systemd-src=file:///path/to/local/clone` on a local clone without network access successfully for a while. Is that good enough, or do we really need to support non-git repos locally?

dm0- commented:
This is for building RPMs where you'd only have the source archive, so yes, it is best to not require Git.

fabiokung commented:
My custom stage1-src flavor build works fine with this. Only two nits:

* `--with-stage1-systemd-revision` is a noop when the systemd src dir is not a git repo. Add a few sentences explaining this on `./configure --help` (or docs elsewhere)?
* `make clean` will wipe your local systemd directory, which bit me a few times as I was pointing to a local source directory I was working on. Consider disabling the `rm -rf ${RKT_STAGE1_SYSTEMD_SRC}` when it is a local dir, or copy contents elsewhere for the build.

Otherwise LGTM.

dm0- commented:
I've added a note to the revision option description, and I've changed the Makefile to copy the local directory instead of using it directly.  This should handle `make clean` properly, and it's probably better for using non-temporary systemd source paths anyway, since the build system would write additional files into the source directory.

fabiokung commented:
cool, re-tested and LGTM.

I should mention that I didn't have any custom patch files being applied to systemd, so I didn't test that. But glancing at the code applying them, it looks fine to me.
